sources/api/netdog/src/wicked.rs

```diff
@@ -0,0 +1,280 @@
+    fn from(dhcp4: Dhcp4) -> Self {
+        match dhcp4 {
+            Dhcp4::DhcpEnabled(b) => WickedDhcp4 {
+                enabled: Some(b),
+                _f: (),
+                ..Default::default()
+            },
+            Dhcp4::WithOptions(o) => WickedDhcp4 {
+                enabled: o.enabled,
+                route_priority: o.route_metric,
+                _f: (),
+            },
+        }
+    }
+}
+
+// The default configuration for AWS based variants
+#[cfg(any(bottlerocket_platform = "aws", bottlerocket_platform = "vmware"))]
+impl Default for WickedInterface {
+    fn default() -> Self {
+        WickedInterface {
+            name: DEFAULT_INTERFACE.to_string(),
+            control: WickedControl::default(),
+            ipv4: Some(WickedIpv4 {
+                arp_verify: Some(false),
+                arp_notify: Some(false),
+                _f: (),
+            }),
+            ipv4_dhcp: Some(WickedDhcp4 {
+                enabled: Some(true),
+                _f: (),
+                ..Default::default()
+            }),
+            ipv6_dhcp: Some(WickedDhcp6 {
+                enabled: Some(true),
+                defer_timeout: Some(1),
+                flags: Some("<optional />".to_string()),
+                _f: (),
+            }),
+        }
+    }
+}
+
+// The default configuration for non-AWS based variants
+#[cfg(bottlerocket_platform = "metal")]
+impl Default for WickedInterface {
+    fn default() -> Self {
+        WickedInterface {
+            name: DEFAULT_INTERFACE.to_string(),
+            control: WickedControl::default(),
+            ipv4: None,
+            ipv4_dhcp: Some(WickedDhcp4 {
+                enabled: Some(true),
+                _f: (),
+                ..Default::default()
+            }),
+            ipv6_dhcp: Some(WickedDhcp6 {
+                enabled: Some(true),
+                defer_timeout: Some(1),
+                flags: Some("<optional />".to_string()),
+                _f: (),
+            }),
+        }
+    }
+}
+
+impl WickedInterface {
+    /// Create a WickedInterface given a name and configuration
+    pub(crate) fn from_config(name: String, config: NetInterface) -> Self {
+        let wicked_dhcp4 = config.dhcp4.map(WickedDhcp4::from);
+        // As additional options are added for IPV6, implement `From` similar to WickedDhcp4
+        let wicked_dhcp6 = config.dhcp6.map(|dhcp6_enabled| WickedDhcp6 {
+            enabled: Some(dhcp6_enabled),
+            _f: (),
+            ..Default::default()
+        });
+        WickedInterface {
+            name,
+            control: WickedControl::default(),
+            ipv4: None,
+            ipv4_dhcp: wicked_dhcp4,
+            ipv6_dhcp: wicked_dhcp6,
+        }
+    }
+
+    /// Serialize the interface's configuration file
+    // Consume `self` to enforce that changes aren't made to the interface type after it has been
+    // written to file
+    pub(crate) fn into_config_file(self) -> Result<()> {
```

Comment:
`into`, generally implies that a converted type will be returned. Also the function becomes test-able if it takes the path and extension as input.

```suggestion
    pub(crate) fn write_config_file(self, config_dir: P, extention: S) -> Result<()> {
```